generic/secrets/security/detected-aws-secret-access-key.yaml

```diff
@@ -1,7 +1,9 @@
 rules:
   - id: detected-aws-secret-access-key
-    pattern-regex: |-
-      (("|'|`)?((?i)aws)_?\w*((?i)secret)_?\w*("|'|`)?\s{0,50}(:|=>|=)\s{0,50}("|'|`)?[A-Za-z0-9/+=]{40}("|'|`)?)
+    patterns:
+    - pattern-regex: |-
+        (("|'|`)?((?i)aws)_?\w*((?i)secret)_?\w*("|'|`)?\s{0,50}(:|=>|=)\s{0,50}("|'|`)?[A-Za-z0-9/+=]{40}("|'|`)?)
+    - pattern-not-regex: EXAMPLE|example|SAMPLE|sample|TEST|test|FAKE|fake|YOURKEY|yourkey|YOURSECRET|yoursecret
```

Comment:
is there a significant performance gain here for this vs `(?i)example|sample|...`?

Reply:
Strangely, when I try (?i)word1|word2 it doesn't work. I was (and am) very puzzled by this.

Reply:
OK... now it's miraculously working again 🤔  I'll use the better version